### PR TITLE
chore: Store a notification to database immediately when it pops up

### PR DESCRIPTION
**Pull request by FeiWang1119**
store a notification to database immediately when it pops up. remove notification from database when it is processed.

Log: Store a notification to database immediately when it pops up

### CODE REVIEW

File: dde-osd/src/notification/bubblemanager.cpp [hunk 1]
```diff
@@ -542,54 +554,42 @@ void BubbleManager::ReplaceBubble(bool replace)
...
     qCDebug(notifiyBubbleLog) << "HandleBubbleEnd, type:" << type << ", bubbleId:" << id
-                              << bubbleParams << selectedHints;
+                              << dbid << selectedHints;
     switch (type) {
     case BubbleManager::Expired:
     case BubbleManager::Unknown:
     case BubbleManager::Dismissed: {
         Q_EMIT NotificationClosed(id, type);
     } break;
+    case BubbleManager::NotProcessedYet: {
```
Comment (18202781743) [inline dde-osd/src/notification/bubblemanager.cpp:567]: 不需要从数据库构造出entity,直接发送bubblemanager的RecordAdded

File: dde-osd/src/notification/bubblemanager.cpp [hunk 2]
```diff
@@ -542,54 +554,42 @@ void BubbleManager::ReplaceBubble(bool replace)
...
-    case BubbleManager::NotProcessedYet: {
-        const auto extraParams = qdbus_cast<QVariantMap>(bubbleParams["extraParams"]);
-        const auto isShowInNotifyCenter = extraParams["isShowInNotifyCenter"].toBool();
-        if (isShowInNotifyCenter) {
-            auto entity = bubbleFromVariantMap(bubbleParams);
-            m_persistence->addOne(entity);
-        }
-    } break;
+    case BubbleManager::Processed: {
+        m_persistence->removeOne(QString::number(dbid));
```
Comment (18202781743) [inline dde-osd/src/notification/bubblemanager.cpp:591]: 如果交互都是QString的话，dbid直接定义为QString类型的，另外dbid看要不要换个名字，之后也不一定用数据库存储，它是notify的id,

File: dde-osd/src/notification/bubble.h [hunk 3]
```diff
@@ -99,5 +100,6 @@ private Q_SLOTS:
     bool m_canClose = false;
     int m_bubbleIndex;
     bool m_beforeLocked;
+    uint m_dbid = 0;
```
[the remark this answers is not in the document]
Comment (FeiWang1119) [reply dde-osd/src/notification/bubble.h]: done

File: dde-osd/src/notification/bubblemanager.cpp (hunk 1 above)
Comment (FeiWang1119) [reply dde-osd/src/notification/bubblemanager.cpp:567]: done

File: dde-osd/src/notification/bubblemanager.cpp (hunk 2 above)
Comment (FeiWang1119) [reply dde-osd/src/notification/bubblemanager.cpp:585]: 统一定义为uint

Comment (FeiWang1119) [reply dde-osd/src/notification/bubblemanager.cpp:585]: 定义为uint

File: dde-osd/src/notification/bubblemanager.cpp [hunk 4]
```diff
@@ -253,11 +258,17 @@ uint BubbleManager::Notify(const QString &appName, uint replacesId,
...
     } else {
-        QVariantMap params;
-        params["isShowPreview"] = enablePreview;
-        params["isShowInNotifyCenter"] = showInNotifyCenter;
-        qCDebug(notifiyBubbleLog) << "Publish ShowBubble, replaceId:" << notification->replacesId();
-        Q_EMIT ShowBubble(appName, replacesId, appIcon, summary, body, actions, hints, expireTimeout, params);
+        if (!useBuiltinBubble()) {
+            QVariantMap params;
+            params["isShowPreview"] = enablePreview;
+            params["isShowInNotifyCenter"] = showInNotifyCenter;
```
Comment (18202781743) [inline dde-osd/src/notification/bubblemanager.cpp:264]: 这个参数，dde-shell没用到吧，

Comment (FeiWang1119) [reply dde-osd/src/notification/bubblemanager.cpp:264]: 删掉 done

File: dde-osd/src/notification/persistence.cpp [hunk 5]
```diff
@@ -112,10 +112,10 @@ void Persistence::addOne(EntityPtr entity)
     } else {
         m_query.next();
         entity->setId(m_query.value(0).toString());
```
[the remark this answers is not in the document]
Comment (FeiWang1119) [reply dde-osd/src/notification/persistence.cpp]: done

File: dde-osd/src/notification/bubblemanager.cpp [hunk 6]
```diff
@@ -871,9 +866,11 @@ Bubble *BubbleManager::createBubble(EntityPtr notify, int index)
     connect(bubble, &Bubble::expired, this, &BubbleManager::bubbleExpired);
     connect(bubble, &Bubble::dismissed, this, &BubbleManager::bubbleDismissed);
     connect(bubble, &Bubble::actionInvoked, this, &BubbleManager::bubbleActionInvoked);
+    connect(bubble, &Bubble::processed, this, [this](EntityPtr ptr){
+        m_persistence->removeOne(ptr->storageId());
+    });
     connect(bubble, &Bubble::notProcessedYet, this, [ this ](EntityPtr ptr) {
-        if (ptr->isShowInNotifyCenter())
-            m_persistence->addOne(ptr);
+        Q_EMIT RecordAdded(ptr->storageId());
```
Comment (18202781743) [inline dde-osd/src/notification/bubblemanager.cpp:873]: 这里需要判断是entity否在通知中心内吧，

Comment (FeiWang1119) [reply dde-osd/src/notification/bubblemanager.cpp:885]: done

**Review by 18202781743 — APPROVED** [review]

**Pull request merged**